### PR TITLE
Add RichErrors to Utils

### PR DESCRIPTION
**Pull request by jalextowle**
## Description

This change adds `RichErrors` to the `utils/` directory. Specifically, `RichErrors` were added to the `Ownable` and `SafeMath` types.

## Testing instructions

1. `yarn build:contracts`
2. `yarn test:contracts`
3. `yarn lint:contracts`

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] Prefix PR title with `[WIP]` if necessary.
-   [x] Add tests to cover changes as needed.
-   [x] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/utils/contracts/src/MixinSafeMathRichErrors.sol [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    // bytes4(keccak256("Uint256OverflowError(uint256,uint256)"))
+    bytes4 internal constant UINT256_OVERFLOW_ERROR =
+        0x55101607;
+
+    // bytes4(keccak256("Uint256UnderflowError(uint256,uint256)"))
+    bytes4 internal constant UINT256_UNDERFLOW_ERROR =
+        0x60ee612f;
+
+    // solhint-disable func-name-mixedcase
+    function Uint256OverflowError(
```
Comment (abandeali1) [inline contracts/utils/contracts/src/MixinSafeMathRichErrors.sol]: These SafeMath errors don't differentiate between addition and multiplication. What if we only had a single `SafeMathError` that took a `SafeMathErrorCode` in addition to `a` and `b`?

```
enum SafeMathErrorCodes {
    UINT256_MULTIPLICATION_OVERFLOW,
    UINT256_ADDITION_OVERFLOW,
    UINT256_SUBTRACTION_UNDERFLOW
}

function SafeMathError(
    SafeMathErrorCodes errorCode,
    uint256 a,
    uint256 b
)
````

Comment (jalextowle) [reply contracts/utils/contracts/src/MixinSafeMathRichErrors.sol]: That makes sense to me. I'll update that.

File: contracts/exchange/test/internal.ts [hunk 2]
```diff
@@ -51,12 +50,12 @@ const emptySignedOrder: SignedOrder = {
     signature: '',
 };
 
-const overflowErrorForCall = new Error(RevertReason.Uint256Overflow);
+const safeMathErrorForCall = () => new SafeMathRevertErrors.SafeMathError();
```
Comment (dorothy-zbornak) [inline contracts/exchange/test/internal.ts]:
```suggestion
const safeMathErrorForCall = new SafeMathRevertErrors.SafeMathError();
```
Should work, right?

File: contracts/exchange/test/internal.ts [hunk 3]
```diff
@@ -51,12 +50,12 @@ const emptySignedOrder: SignedOrder = {
...
 
 describe('Exchange core internal functions', () => {
     let chainId: number;
     let testExchange: TestExchangeInternalsContract;
-    let overflowErrorForSendTransaction: Error | undefined;
+    let safeMathErrorForSendTransaction: () => Error | undefined;
```
Comment (dorothy-zbornak) [inline contracts/exchange/test/internal.ts]:
```suggestion
    let safeMathErrorForSendTransaction: () => RevertError | undefined;
```

File: contracts/utils/contracts/src/MixinSafeMathRichErrors.sol [hunk 4]
```diff
@@ -0,0 +1,35 @@
+pragma solidity ^0.5.9;
+
+import "./RichErrors.sol";
+
+
+contract MixinSafeMathRichErrors is
+    RichErrors
+{
+    // bytes4(keccak256("SafeMathError(uint8,uint256,uint256)"))
+    bytes4 internal constant SAFE_MATH_ERROR =
```
Comment (dorothy-zbornak) [inline contracts/utils/contracts/src/MixinSafeMathRichErrors.sol]:
```suggestion
    bytes4 internal constant SAFE_MATH_ERROR_SELECTOR =
```

File: contracts/utils/contracts/src/MixinOwnableRichErrors.sol [hunk 5]
```diff
@@ -0,0 +1,28 @@
+pragma solidity ^0.5.9;
+
+import "./RichErrors.sol";
+
+
+contract MixinOwnableRichErrors is
+    RichErrors
+{
+    // bytes4(keccak256("OnlyOwnerError(address,address)"))
+    bytes4 internal constant ONLY_OWNER_SELECTOR =
```
Comment (dorothy-zbornak) [inline contracts/utils/contracts/src/MixinOwnableRichErrors.sol]:
```suggestion
    bytes4 internal constant ONLY_OWNER_ERROR_SELECTOR =
```

**Pull request merged**